src/bench/pqs/spraylist.cpp

```diff
@@ -55,14 +55,38 @@ void
 spraylist::insert(const uint32_t &k,
                   const uint32_t &v)
 {
-    sl_add(m_q, k, READ_ADD_REM_ELASTIC_TX);
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
+}
+
+void
+spraylist::insert(const size_t &k,
+                  const size_t &v)
+{
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
 }
 
 bool
-spraylist::delete_min(uint32_t &v)
+spraylist::delete_min(size_t &k, size_t &v)
 {
-    const int ret = spray_delete_min(m_q, &v, d);
-    return (ret == 1);
+    unsigned long k_ret;
+    unsigned long v_ret;
+    int ret;
+    do {
+        ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
```

Comment:
The src/bench/shortest_paths.cpp benchmark has a stop condition based on a global counter of the number of active tasks, so I think it calculates the correct result regardless of spurious failures - is that not the case? See https://github.com/klsmpq/klsm/blob/master/src/bench/shortest_paths.cpp#L192, a failure just results in retrying the deletion.

Anyway, if a non-spurious failure mode does turn out to be required, maybe we should add it as another method to avoid influencing existing benchmarks.

Once this question is resolved, it's all good to land from my side :+1: